templates/custom_data.sh.tpl

```diff
@@ -139,18 +171,78 @@ function directory_create {
+  sudo curl -Os https://releases.hashicorp.com/"$${PRODUCT}"/"$${VERSION}"/"$${PRODUCT}"_"$${VERSION}"_SHA256SUMS
+	log "INFO" "Downloading $${PRODUCT}  Enterprise binary checksum signature file"
+  sudo curl -Os https://releases.hashicorp.com/"$${PRODUCT}"/"$${VERSION}"/"$${PRODUCT}"_"$${VERSION}"_SHA256SUMS.sig
+  log "INFO" "Verifying the signature file is untampered."
+  gpg --verify "$${PRODUCT}"_"$${VERSION}"_SHA256SUMS.sig "$${PRODUCT}"_"$${VERSION}"_SHA256SUMS
+	if [[ $? -ne 0 ]]; then
+		log "ERROR" "Gpg verification failed for SHA256SUMS."
+		exit_script 1
+	fi
+  if [ -x "$(command -v sha256sum)" ]; then
+		log "INFO" "Using sha256sum to verify the checksum of the $${PRODUCT} binary."
+		sha256sum -c "$${PRODUCT}"_"$${VERSION}"_SHA256SUMS --ignore-missing
+	else
+		log "INFO" "Using shasum to verify the checksum of the $${PRODUCT} binary."
+		shasum -a 256 -c "$${PRODUCT}"_"$${VERSION}"_SHA256SUMS --ignore-missing
+	fi
+	if [[ $? -ne 0 ]]; then
+		log "ERROR" "Checksum verification failed for the $${PRODUCT} binary."
+		exit_script 1
+	fi
+
+	log "INFO" "Checksum verification passed for the $${PRODUCT} binary."
+
+	log "INFO" "Removing the downloaded files to clean up"
+	sudo rm -f "$${PRODUCT}"_"$${VERSION}"_SHA256SUMS "$${PRODUCT}"_"$${VERSION}"_SHA256SUMS.sig
+
+}
+
 # install_vault_binary downloads the Vault binary and puts it in dedicated bin directory
 function install_vault_binary {
-  log "INFO" "Downloading Vault Enterprise binary"
-  sudo curl -so $VAULT_DIR_BIN/vault.zip $VAULT_INSTALL_URL
+  local OS_ARCH="$1"
 
-  log "INFO" "Unzipping Vault Enterprise binary to $VAULT_DIR_BIN"
-  sudo unzip $VAULT_DIR_BIN/vault.zip vault -d $VAULT_DIR_BIN
-  sudo unzip $VAULT_DIR_BIN/vault.zip -x vault -d $VAULT_DIR_LICENSE
+  log "INFO" "Deploying Vault Enterprise binary to $VAULT_DIR_BIN unzip and set permissions"
+  sudo unzip "$${PRODUCT}"_"$${VAULT_VERSION}"_"$${OS_ARCH}".zip  vault -d $VAULT_DIR_BIN
+  sudo unzip "$${PRODUCT}"_"$${VAULT_VERSION}"_"$${OS_ARCH}".zip -x vault -d $VAULT_DIR_LICENSE
+  sudo rm -f "$${PRODUCT}"_"$${VAULT_VERSION}"_"$${OS_ARCH}".zip
 
-  sudo rm $VAULT_DIR_BIN/vault.zip
+	log "INFO" "Deploying Vault $VAULT_DIR_BIN set permissions"
+  sudo chmod 0755 $VAULT_DIR_BIN/vault
+  sudo chown $VAULT_USER:$VAULT_GROUP $VAULT_DIR_BIN/vault
+
+  log "INFO" "Deploying Vault create symlink "
+  sudo ln -sf $VAULT_DIR_BIN/vault /usr/local/bin/vault
+
+  log "INFO" "Vault binary installed successfully at $VAULT_DIR_BIN/vault"
 }
 
+# # install_vault_binary downloads the Vault binary and puts it in dedicated bin directory
+# function install_vault_binary {
+#   log "INFO" "Downloading Vault Enterprise binary"
+#   sudo curl -so $VAULT_DIR_BIN/vault.zip $VAULT_INSTALL_URL
+
+#   log "INFO" "Unzipping Vault Enterprise binary to $VAULT_DIR_BIN"
+#   sudo unzip $VAULT_DIR_BIN/vault.zip vault -d $VAULT_DIR_BIN
+#   sudo unzip $VAULT_DIR_BIN/vault.zip -x vault -d $VAULT_DIR_LICENSE
+
+#   sudo rm $VAULT_DIR_BIN/vault.zip
+# }
+
```

Comment:
Is the proposal to remove this block? Should it actually be removed as a result?

Reply:
yeah, totally an oversight on last update will delete and push